### PR TITLE
Json fixups/v2

### PR DESCRIPTION
**Pull request by victorjulien**
As #10132, but with ticket reference for:
https://redmine.openinfosecfoundation.org/issues/6664

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17407

**Review by jasonish — APPROVED** [review]
Looks OK to me.

File: src/output-json-http.c [hunk 1]
```diff
@@ -288,20 +288,25 @@ static void EveHttpLogJSONExtended(JsonBuilder *js, htp_tx_t *tx)
                 js, "protocol", bstr_ptr(tx->request_protocol), bstr_len(tx->request_protocol));
     }
 
-    /* response status */
-    if (tx->response_status != NULL) {
+    /* response status: from libhtp:
+     * "Response status code, available only if we were able to parse it, HTP_STATUS_INVALID
+     *  otherwise. HTP_STATUS_UNKNOWN until parsing is attempted" .*/
+    const int resp = tx->response_status_number;
+    if (resp > 0) {
```
Comment (catenacyber) [inline src/output-json-http.c:295]: Why not use `HTP_STATUS_UNKNOWN` as documented ?

**Comment by victorjulien** [comment]
Merged in #10134, thanks!

**Pull request closed without merging**